pkg/api/schema/schema.proto

```diff
@@ -94,10 +94,22 @@ message OpenSessionResponse {
 
 ////////////////////////////////////////////////////////
 
+message KVConstraints {
+	// If set to true, only store new value if existing one with same key already exists
+	bool mustExist = 1;
+
+	// If set to true, only store new value if value with same key does not exist yet
+	bool mustNotExist = 2;
+
+	// If set to value > 0, only store new value if the existing value does not exist or was not modified after given transaction
+	uint64 notModifiedAfterTX = 3;
+}
+
 message KeyValue {
 	bytes key = 1;
 	bytes value = 2;
 	KVMetadata metadata = 3;
+	KVConstraints constraints = 4;
```

Comment:
probably constraints could be at SetRequest... it'd be even possible to set a constraint over a key which is not inserted or updated but read... It may add a bit of complexity on the message types but on the other hand it would be equivalent to constraints in read-write transactions... wdyt?

Reply:
Yes, that's a very interesting idea, the only drawback I could see here is that we would have to send key values twice for constrained entries - if keys were long, that would mean a significant increase of transferred data. Should we worry about that?

Reply:
for the constraints it may be possible to use the hash of the key...